app/controllers/concerns/report_embeddable.rb

```diff
@@ -32,7 +33,10 @@ def build_report_data(options = {})
   # returns true if no errors, false otherwise
   def run_and_handle_errors
     begin
-      @report.run(current_ability)
+      Rails.cache.fetch(@report.cache_key_with_responses(current_mission)) do
+        @report.run(current_ability)
+        @report
+      end
```

Comment:
I'm not super happy with the structure here. It seems like line 5 should not be necessary.

How about this: 
* we rename this method to run_or_fetch_and_handle_errors`
* add a comment saying `# Looks for a cached, populated report object matching @report. If one is found, stores it in @report. If not found, calls run on the existing @report`
* add `@report = ` to the front of line 36.

Does this work? Then we can remove line 5 and a lot of confusion (at least for me!)

We can also get rid of return true and return false. I don't think they are used.